parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java

```diff
@@ -140,8 +140,9 @@ public void close() throws IOException, InterruptedException {
         }
         finalMetadata.putAll(finalWriteContext.getExtraMetaData());
         parquetFileWriter.end(finalMetadata);
+        AutoCloseables.uncheckedClose(parquetFileWriter);
```

Comment:
`parquetFileWriter.end(Map<String, String>)` invokes `parquetFileWriter.close()`, so this call is not necessary. Why we added the `close()` call to `finally` to ensure the related buffers are freed (required for direct memory handling).
Maybe introduce a similar logic of having an `aborted` flag in `ParquetFileWriter` just like we have done it for `InternalParquetRecordWriter`.